src/InteractiveBridge.ts

```diff
@@ -207,82 +226,183 @@ export class InteractiveBridge extends LiveryBridge {
+   * and calls back `listener` with any subsequent errors.
+   */
+  subscribeError(listener: (value: string | undefined) => void) {
+    return this.sendCommand('subscribeError', undefined, (value) =>
+      listener(validateStringOrUndefined(value)),
+    ).then(validateStringOrUndefined);
+  }
+
   /**
    * Returns promise of current LiveryPlayer fullscreen state
    * and calls back `listener` with any subsequent state changes.
+   *
+   * @deprecated Instead use {@link subscribeDisplay}.display value "FULLSCREEN"
    */
   subscribeFullscreen(listener: (value: boolean) => void) {
-    function validate(value: unknown) {
-      if (typeof value !== 'boolean') {
-        throw new Error(
-          `subscribeFullscreen value type: ${typeof value}, should be: boolean`,
-        );
-      }
-      return value;
-    }
-
     return this.sendCommand('subscribeFullscreen', undefined, (value) =>
-      listener(validate(value)),
-    ).then(validate);
+      listener(validateBoolean(value)),
+    ).then(validateBoolean);
+  }
+
+  /**
+   * Returns promise of current mode of playback, e.g. how to buffer, sync, adapt quality, manage stalls, etc.
+   * and calls back `listener` with any subsequent mode changes.
+   */
+  subscribeMode(listener: (mode: PlaybackMode) => void) {
+    return this.sendCommand('subscribeMode', undefined, (mode) =>
+      listener(playbackMode.parse(mode)),
+    ).then((mode) => playbackMode.parse(mode));
+  }
+
+  /**
+   * Returns promise of current LiveryPlayer muted state
+   * and calls back `listener` with any subsequent muted changes.
+   */
+  subscribeMuted(listener: (muted: boolean) => void) {
+    return this.sendCommand('subscribeMuted', undefined, (value) =>
+      listener(validateBoolean(value)),
+    ).then(validateBoolean);
   }
 
   /**
    * Returns promise of current LiveryPlayer window orientation (`'landscape' \| 'portrait'`)
    * and calls back `listener` with any subsequent orientations.
+   *
+   * @deprecated Will be removed in the next major version.
    */
-  subscribeOrientation(listener: (orientation: Orientation) => void) {
-    function validate(value: unknown) {
-      if (value !== 'landscape' && value !== 'portrait') {
-        const strValue = stringify(value);
-        throw new Error(
-          `subscribeOrientation value: ${strValue}, should be: "landscape" | "portrait"`,
-        );
-      }
-      return value;
-    }
-
+  subscribeOrientation(listener: (value: Orientation) => void) {
     return this.sendCommand('subscribeOrientation', undefined, (value) =>
-      listener(validate(value)),
-    ).then(validate);
+      listener(orientation.parse(value)),
+    ).then((value) => orientation.parse(value));
+  }
+
+  /**
+   * Returns promise of current playback paused state
+   * and calls back `listener` with any subsequent paused state updates.
+   */
+  async subscribePaused(listener: (value: boolean) => void) {
```

Comment:
Using Zod for this seems needlessly complicated to me, also you are calling the listener each time `playbackState` has changed, not just when `paused` has changed.

Please use the `reducedSubscribe()` util function as described in my AbstractPlayerBridge comment above here as well and with a simple reducer function like `['ENDED', 'PAUSED'].includes(value)`.

Reply:
I don't necessarily agree with not using Zod here, as the difference in code would be similar when not using Zod and it makes sense to me to try to keep things consistent in how the bridge subscriptions are validated.

Reply:
But this is not validation, this is reduction. Validation has already been taken care of within `subscribePlaybackstate`. And when using Zod for that purpose as well you would end up with a schema instance and a reduction function based on that with those being in another file then where they are used. Instead you could simply inline the reducer function here and match it up with the functional description in the TSDoc. The other schema's are actually used for validation and their types can also be used for player bridge implementation, interactive bridge usage and documentation. Unless you can think of another use for the PausedState schema and reducer besides here this is just a single use thing.

Reply:
I was trying to avoid defining literals in two different places but I will do what you're asking.